src/lib/shared_utils/src/common/types/utility_token/token_event.rs

```diff
@@ -15,7 +15,16 @@ pub enum TokenEvent {
         timestamp: SystemTime,
     },
     Burn,
-    Transfer,
+    Transfer {
+        amount: u64,
+        to_account: Principal,
+        timestamp: SystemTime,
+    },
+    Receive {
+        amount: u64,
+        from_account: Principal,
+        timestamp: SystemTime,
+    },
```

Comment:
I think lets use more descriptive name here. `HotorNotTransfer` and `HotorNotRecieve`